aspnet/RVTR.Lodging.WebApi/Controllers/LodgingController.cs

```diff
@@ -1,110 +1,139 @@
+        ///
+        /// </summary>
+        /// <param name="logger"></param>
+        /// <param name="unitOfWork"></param>
+        public LodgingController(ILogger<LodgingController> logger, UnitOfWork unitOfWork)
+        {
+            _logger = logger;
+            _unitOfWork = unitOfWork;
+        }
 
-        return Ok();
-      }
-      catch
-      {
-        return NotFound(id);
-      }
-    }
 
-    /// <summary>
-    ///
-    /// </summary>
-    /// <returns></returns>
-    [HttpGet]
-    public async Task<IActionResult> Get()
-    {
-      return Ok(await _unitOfWork.Lodging.SelectAsync());
-    }
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        [HttpDelete("{id}")]
+        public async Task<IActionResult> Delete(int id)
+        {
+            try
+            {
+                await _unitOfWork.Lodging.DeleteAsync(id);
+                await _unitOfWork.CommitAsync();
 
-    /// <summary>
-    ///
-    /// </summary>
-    /// <param name="id"></param>
-    /// <returns></returns>
-    [HttpGet("{id}")]
-    public async Task<IActionResult> Get(int id)
-    {
-      try
-      {
-        return Ok(await _unitOfWork.Lodging.SelectAsync(id));
-      }
-      catch
-      {
-        return NotFound(id);
-      }
-    }
+                return Ok();
+            }
+            catch
+            {
+                return NotFound(id);
+            }
+        }
 
-    /// <summary>
-    ///
-    /// </summary>
-    /// <param name="lodging"></param>
-    /// <returns></returns>
-    [HttpPost]
-    public async Task<IActionResult> Post(LodgingModel lodging)
-    {
-      await _unitOfWork.Lodging.InsertAsync(lodging);
-      await _unitOfWork.CommitAsync();
+        /// <summary>
+        ///
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<IActionResult> Get()
+        {
+            var lodgings = await _unitOfWork.Lodging.SelectAsync();
+            if (lodgings == null)
+                return NotFound();
+            else
+                return Ok(lodgings);
+        }
 
-      return Accepted(lodging);
-    }
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        [HttpGet("{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<IActionResult> Get(int id)
+        {
+            if (id <= 0)
+            {
+                return NotFound(id);
+            }
+            else
+            {
+                try
+                {
+                    return Ok(await _unitOfWork.Lodging.SelectAsync(id));
+                }
+                catch
+                {
```

Comment:
What exception are we trying to catch here? I think we have to be more specific when using catch statements. Also, the first if statement is unnecessary because SelectAsync would throw an exception if the id was 0 or less too.

Reply:
removed the try-catch exception handling.